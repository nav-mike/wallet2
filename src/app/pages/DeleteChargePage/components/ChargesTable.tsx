import React from 'react';
import { ChargesTableHeader } from './ChargesTableHeader';
import { ChargesTableRows } from './ChargesTableRows';
import styles from './ChargesTable.module.css';

export interface ICharge {
  date: Date;
  id: string;
  value: number;
  type: string;
}

export interface IChargesTableProps {
  charges: ICharge[];
}

export const ChargesTable = (props: IChargesTableProps) => {
  return (
    <table className={styles['charge-table__table-component']}>
      <thead>
        <ChargesTableHeader />
      </thead>
      <ChargesTableRows charges={props.charges} />
    </table>
  );
};
