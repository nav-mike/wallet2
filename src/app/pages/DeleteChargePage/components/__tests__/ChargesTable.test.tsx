import * as React from 'react';
import { render } from '@testing-library/react';
import { ChargesTable } from '../ChargesTable';

describe('<ChargesTable />', () => {
  const data = [{ date: new Date(), id: '123-234', value: 120, type: 'food' }];

  it('should render and match the snapshot', () => {
    const table = render(<ChargesTable charges={data} />);
    expect(table.container.firstChild).toMatchSnapshot();
  });
});
