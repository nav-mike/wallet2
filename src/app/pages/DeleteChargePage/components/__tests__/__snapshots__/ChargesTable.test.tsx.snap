// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChargesTable /> should render and match the snapshot 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: rgba(220,120,95,1);
  border: 1px solid rgba(220,120,95,1);
  border-radius: 0.5rem;
  min-height: 2rem;
}

.c0:hover {
  cursor: pointer;
  background-color: #fab9a9;
}

<table
  class="charge-table__table-component"
>
  <thead>
    <tr>
      <th>
        Date
      </th>
      <th>
        Value
      </th>
      <th>
        Type
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        3/16/2021
      </td>
      <td>
        120
      </td>
      <td>
        food
      </td>
      <td>
        <button
          class="c0"
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
